### PR TITLE
Add Material You colour scheme

### LINKED ISSUE #22: Add material You
There is currently only dark and light theme but no Material You theme

### PR DESCRIPTION
**Pull request by DreamyLynn**
This is not yet ready to merge, but I'm working on this and had to put a new Interface for preference support requirements in place, since Material You is not supported on API levels<31. Please check whether you are okay with the implementation I used for this. Thank you!

Closes #22

### CODE REVIEW

**Comment by Fxsch** [comment]
Is this still being worked on and what is missing for this to be finished?

**Comment by DreamyLynn** [comment]
This is on hold for now, as I have not received a response from the maintainers regarding the implementation yet. I'm happy to start work on this again, once this is approved/a different way is proposed. Finishing this should only be a matter of finding the right attribute values and testing iirc (it's been a while 😅).

**Comment by jsixface** [comment]
@DreamyLynn I am not an UI expert. This looks okay to me. If you have tested and it works, can you make this PR ready for review?

**Comment by DreamyLynn** [comment]
I'll get it ready as soon as I can, still have to do a few tweaks and proper testing.

**Comment by nofishonfriday** [comment]
@DreamyLynn Are you still planning to finalize this PR? I hope so. :)

**Comment by DreamyLynn** [comment]
Yeah I am, I have a lot of stress right now so I barely have time for this 😅. I will finish it when I can, sorry for the wait.

**Comment by DreamyLynn** [comment]
Hey, it's been a while :). Sorry for taking so long, I'm finally done with my exams and had time for proper testing, everything is working nicely.

There's still one edge case, if you do a Seedvault/adb backup of the app with Material You on (On Android > S), then restore that on Android lower than S, the app will crash on launch. Since this is highly unlikely to happen I don't think it's worth the check. Otherwise it's good to go.

**Comment by ThatOneCalculator** [comment]
If anyone's restoring a seedvault backup from a 12+ version of android on an android version before 12, they probably have bigger problems to worry about :smile:

**Comment by ThatOneCalculator** [comment]
Also, any screenshots we could take a look at? :eyes:

**Comment by DreamyLynn** [comment]
-204239.png](https://github.com/quillpad/quillpad/assets/59802880/75772f5d-12

t_20240308-204248.png …]()

**Comment by DreamyLynn** [comment]
Actually done now.
Some Screenshots:
<details>

<summary>Purple</summary>

Screenshot_20240308-215449
Screenshot_20240308-215452

</details>

<details>

<summary>Blue</summary>

Screenshot_20240308-215521
Screenshot_20240308-215519

</details>

<details>

<summary>Orange</summary>


Screenshot_20240308-215512
Screenshot_20240308-215510

</details>

<details>

<summary>Green</summary>

Screenshot_20240308-215503
Screenshot_20240308-215501

</details>

**Comment by jsixface** [comment]
> Actually done now. Some Screenshots:
> Purple
> Blue
> Orange
> Green

Are these generated by setting the System UI to these colors? I see the color scheme preference in the app also have these colors predefined. I just wanted to make sure these are System theme.

File: app/src/main/java/org/qosp/notes/ui/settings/SettingsUtils.kt [hunk 1]
```diff
@@ -16,6 +18,9 @@ inline fun <reified T> Fragment.showPreferenceDialog(
     val enumValues = enumValues<T>()
     val selectedIndex = enumValues.indexOf(selected)
     val items = items ?: enumValues
+        .filter {
+            !(it is HasSupportRequirement && !it.isSupported())
```
Comment (jsixface) [inline app/src/main/java/org/qosp/notes/ui/settings/SettingsUtils.kt:22]: how about 
```
.filter {
            it !is HasSupportRequirement || it.isSupported()
```

**Comment by jsixface** [comment]
Can you also rebase your branch off of `quillpad:master`?

**Comment by jsixface** [comment]
> Yes, these are the system-generated UI colours as in the colours you select when setting your wallpaper. Implemented your suggestion, rebased and squashed.

Thank you so much.

**Comment by jsixface** [comment]
@DreamyLynn The build is failing. Can you take a look - https://github.com/quillpad/quillpad/actions/runs/8232997702/job/22516913803

**Comment by DreamyLynn** [comment]
Sorry, made a mistake while resolving a merge conflict.

**Comment by RustoMCSpit** [comment]
this is so cool thank you so much

**Comment by RustoMCSpit** [comment]
how do you apply it

**Pull request merged**